### PR TITLE
Fix PHP 7 incompatibilities

### PR DESCRIPTION
**Pull request by emirb**
- Class Void renamed to VoidAdapter
- Updated Guzzle to 5.3.1 due to invalid syntax (https://github.com/guzzle/guzzle/pull/1393)
- Fixed outdated composer.lock

### CODE REVIEW

**Comment by emirb** [comment]
Can you please add Travis (or similar) continuous integration service to the repository?

File: src/schibsted/payment/sdk/adapters/Curl.php [hunk 1]
```diff
@@ -18,9 +18,9 @@ class Curl extends \schibsted\payment\lib\sdk\Adapter
     /**
      * Makes an HTTP request.
      *
-     * @param String $uri the URI to make the request to
-     * @param Array $post the parameters to use for the POST body
-     * @param CurlHandler $ch optional initialized curl handle
+     * @param string $uri the URI to make the request to
+     * @param array $post the parameters to use for the POST body
+     * @param \GuzzleHttp\Ring\Client\CurlHandler $ch optional initialized curl handle
```
Comment (alkemann) [inline src/schibsted/payment/sdk/adapters/Curl.php]: This is an actual Curl adapter that is not related to Guzzle. So this is an invalid change

File: src/schibsted/payment/lib/Resource.php [hunk 2]
```diff
@@ -16,6 +16,7 @@
     protected $_sdk = null;
 
     protected $_sdk_class = '';
+    protected $name;
```
Comment (alkemann) [inline src/schibsted/payment/lib/Resource.php]: $name field has to be, and is, set on the sub classes. Abstract field, if there was such a thing.

Comment (emirb) [reply src/schibsted/payment/lib/Resource.php]: This was mostly for IDE complaints :-)

File: src/schibsted/payment/sdk/adapters/Curl.php (hunk 1 above)
Comment (emirb) [reply src/schibsted/payment/sdk/adapters/Curl.php]: Can you please point me to the actual file where the adapter is defined?

Comment (alkemann) [reply src/schibsted/payment/sdk/adapters/Curl.php]: THIS file.. the one you changed and we are commenting on :)

Comment (emirb) [reply src/schibsted/payment/sdk/adapters/Curl.php]: Sorry, my bad. I was confused by the fact that CurlHandler class does not exist. If it existed then it'd be a violation of PSR-0 autoloading.  

By your comment, the parameter should be changed to `\schibsted\payment\sdk\adapters\Curl`. However, the thing is that the the parameter will actually be a resource (check `curl_init` return), so the type-hint against anything except resource would be misleading. As this is just a part of the care-requiring codebase, I'll revert my changes, to not open a Pandora box. :)

Comment (alkemann) [reply src/schibsted/payment/sdk/adapters/Curl.php]: The value that is returned does not have a class, it is type hinted only as "resource" in the php docs : http://php.net/manual/en/function.curl-init.php so I am not sure what typehint would would here. That docblock line was written like that for human to read not IDE I guess.

**Comment by alkemann** [comment]
Ok Ill merge this and look at the issues with travis and the typehinting.

**Comment by emirb** [comment]
Thanks @alkemann!
